Fix broken test in Ledger.d

AndrejMitrovic opened the pull request:
This problem showed up once I started using a tx pool in the Ledger. It started erroring about duplicate transactions. Turns out the test was wrong here.

AndrejMitrovic committed:
Fix test-case in Ledger.d

It was creating duplicate transactions.

It wasn't caught because the Ledger does not
contain the pool of existing transactions,
it currently lives in the GossipProtocol.

Once the TransactionPool is implemented,
the Ledger will be able to check if the
transaction already exists and reject it.

Geod24 commented on source/agora/node/Ledger.d:
```diff
@@ -252,7 +252,8 @@ unittest
 
-        last_txs = txes;
+        // keep track of last tx's to chain them to
+        last_txs = txes[$ - Block.TxsInBlock .. $];;
```
Shouldn't it be:
```suggestion
        last_txs = txes[$ - (Block.TxsInBlock * count) .. $];
```

AndrejMitrovic replied:
I just want the last 8 transactions, regardless of how many are there before that.

Geod24 replied:
Ah right: https://github.com/bpfkorea/agora/blob/f5bed737df2d19a1e83b6bf2ea4b5121f27d7903/source/agora/common/Transaction.d#L348

Maybe the function should reflect that

Geod24 replied:
```suggestion
        last_txs = txes[$ - Block.TxsInBlock .. $];
```

AndrejMitrovic replied:
> Maybe the function should reflect that

You mean to add docs? Sure.

AndrejMitrovic committed:
Add assert in makeChainedTransactions()

The previous transactions should either be null,
or the previous set of transactions that constitutes
an entire block (e.g. 8 transactions currently).

AndrejMitrovic commented:
Added assertion in `getChainedTransactions` and removed double semicolon.

AndrejMitrovic commented:
Green.
